Correct META-INF/LICENSE handling

joakime opened the pull request:
Fixes #5168 

The `jetty-runner` uber-jar now uses `META-INF/LICENSE` from the maven build, not the dependencies it unpacks.

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

joakime committed:
Correct META-INF/LICENSE handling

The uber-jar now uses META-INF/LICENSE from the maven build, not the dependencies.

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

sbordet reviewed (changes requested):
The generated jar still contains a large amount of unwanted files.

sbordet committed:
Issue #5168 - Jetty 10 artifact archives should have EPL 2 license in them.

Excluded unnecessary files from jetty-runner jar.
Moved MANIFEST.MF to proper place to it does not get copied into the jar root.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

sbordet commented:
@joakime please confirm that my changes still generate a valid jar.
There were a lot of unnecessary files that needed to be removed, some still referencing the Eclipse 1.0 license.

olamy commented:
while we are here maybe we should remove all test dependencies? (junit, hamcrest etc...)

olamy commented on jetty-runner/pom.xml:
```diff
@@ -27,7 +27,18 @@
+              </excludes>
               <outputDirectory>${project.build.directory}/classes</outputDirectory>
               <overWriteReleases>false</overWriteReleases>
               <overWriteSnapshots>true</overWriteSnapshots>
```
exclude test dependencies?
```
  <excludeScope>test</excludeScope>
  <includeScope>runtime</includeScope>
```

sbordet commented:
@olamy yes please exclude test dependencies.

olamy committed:
do not include test dependencies in jetty-runner jar

Signed-off-by: olivier lamy <oliver.lamy@gmail.com>
